crates/typst-pdf/src/gradient.rs

```diff
@@ -391,10 +396,16 @@ fn write_patch(
     ]));
 
     let colors =
-        [c0.map(u16::to_be), c0.map(u16::to_be), c1.map(u16::to_be), c1.map(u16::to_be)];
+        [c0.iter().map(u16_to_be),
+         c0.iter().map(u16_to_be),
+         c1.iter().map(u16_to_be),
+         c1.iter().map(u16_to_be)];
 
     // Push the colors.
-    target.extend_from_slice(bytemuck::cast_slice(&colors));
+    // TODO: find a better solution
+    unsafe {
+        target.extend_from_slice(colors.align_to::<u8>().1);
```

Comment:
Please do not do that, there is no reason for unsafe code here. It's better to iterate over the colors and extend for each component than this.

Reply:
Agreed! I've just replaced this part to avoid the unsafe code. (To be improved.)

Reply:
New version without unsafe code:

https://github.com/typst/typst/pull/4568/files#diff-6a20014eeaaa0cd7d097c8ead96193dfc6ed65aea4ed7ff2de0b5584b92dee78L397-R429

```rust
    target.extend_from_slice(
        &c0.iter()
            .flat_map(|arg0: &u16| u16::to_be_bytes(*arg0))
            .cycle()
            .take(4 * c0.len())
            .chain(
                c1.iter()
                    .flat_map(|arg0: &u16| u16::to_be_bytes(*arg0))
                    .cycle()
                    .take(4 * c1.len()),
            )
            .collect::<SmallVec<[u8; 32]>>(),
    );
```

Like for the above changes, I don't have a good feel for what the performance of this is.